webknossos/script-collection/globalize_floodfill.py

```diff
@@ -0,0 +1,364 @@
+
+from webknossos.geometry import BoundingBox, Mag, Vec3Int
+
+BUCKET_SIZE = 32
+BUCKET_SHAPE = (BUCKET_SIZE, BUCKET_SIZE, BUCKET_SIZE)
+CUBE_SHAPE = BUCKET_SHAPE  # (256, 256, 256)
+NEIGHBORS = [(1, 0, 0), (-1, 0, 0), (0, 1, 0), (0, -1, 0), (0, 0, 1), (0, 0, -1)]
+
+FloodFillBbox = namedtuple(
+    "FloodFillBbox",
+    ["bounding_box", "seed_position", "source_id", "target_id", "timestamp"],
+)
+
+
+def create_parser() -> ArgumentParser:
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "--volume_path",
+        "-v",
+        help="Directory containing the volume tracing.",
+        type=Path,
+    )
+
+    parser.add_argument(
+        "--segmentation_layer_path",
+        "-s",
+        help="Directory containing the segmentation layer.",
+        type=Path,
+    )
+
+    parser.add_argument(
+        "--nml_path",
+        "-n",
+        help="NML that contains the bounding boxes",
+        type=Path,
+    )
+
+    parser.add_argument("--output_path", "-o", help="Output directory", type=Path)
+
+    parser.add_argument("--skip_merge", default=False, action="store_true")
+
+    return parser
+
+
+def get_bucket_pos_and_offset(
+    global_position: Tuple[int, int, int]
+) -> Tuple[Tuple[int, int, int], Tuple[int, int, int]]:
+    offset = (
+        global_position[0] % CUBE_SHAPE[0],
+        global_position[1] % CUBE_SHAPE[1],
+        global_position[2] % CUBE_SHAPE[2],
+    )
+    return (
+        substract_positions(global_position, offset),
+        offset,
+    )
+
+
+def add_positions(
+    a: Tuple[int, int, int], b: Tuple[int, int, int]
+) -> Tuple[int, int, int]:
+    return a[0] + b[0], a[1] + b[1], a[2] + b[2]
+
+
+def substract_positions(
+    a: Tuple[int, int, int], b: Tuple[int, int, int]
+) -> Tuple[int, int, int]:
+    return a[0] - b[0], a[1] - b[1], a[2] - b[2]
+
+
+def inside_bbox(
+    point: Tuple[int, int, int],
+    bbox_top_left: Tuple[int, int, int] = (0, 0, 0),
+    bbox_bottom_right: Tuple[int, int, int] = CUBE_SHAPE,
+) -> bool:
+    return (
+        bbox_top_left[0] <= point[0] < bbox_bottom_right[0]
+        and bbox_top_left[1] <= point[1] < bbox_bottom_right[1]
+        and bbox_top_left[2] <= point[2] < bbox_bottom_right[2]
+    )
+
+
+def execute_floodfill(
+    data_path: Path,
+    seed_position: Tuple[int, int, int],
+    already_processed_bbox: BoundingBox,
+    source_id: int,
+    target_id: int,
+) -> None:
+    # bucketData[neighbourVoxelIndex] == sourceCellId || (isInsideBBox & & bucketData[neighbourVoxelIndex] == targetCellId)
+    bucket_and_seed_pos: List[Tuple[Tuple[int, int, int], Tuple[int, int, int]]] = [
+        get_bucket_pos_and_offset(seed_position)
+    ]
+    # bucket_to_seed_pos: Dict[
+    #    Tuple[int, int, int], Set[Tuple[int, int, int]]
+    # ] = defaultdict(set)
+    already_processed_bbox_top_left = already_processed_bbox.topleft.to_tuple()
+    already_processed_bbox_bottom_right = already_processed_bbox.bottomright.to_tuple()
+    visited = np.zeros(
```

Comment:
please add a comment stating that this variable is only necessary for traversing the already processed bbox.